python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedOutboundCommunicationsCheck.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.cdk;
+
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.checks.Expressions;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6463")
+public class UnrestrictedOutboundCommunicationsCheck extends AbstractCdkResourceCheck {
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqns(List.of("aws_cdk.aws_ec2.SecurityGroup", "aws_cdk.aws_ec2.SecurityGroup.from_security_group_id"), (subscriptionContext, callExpression) -> {
+      RegularArgument argument = TreeUtils.nthArgumentOrKeyword(3, "allow_all_outbound", callExpression.arguments());
+      if (argument == null) {
+        subscriptionContext.addIssue(callExpression, "Omitting \"allow_all_outbound\" enables unrestricted outbound communications. Make sure it is safe here.");
+      } else if (isArgumentTrue(argument.expression())) {
+        subscriptionContext.addIssue(argument, "Make sure that allowing unrestricted outbound communications is safe here.");
+      }
+    });
+  }
+
+  private static boolean isArgumentTrue(Expression expression) {
+    if (expression != null && expression.is(Tree.Kind.NAME)) {
+      Name name = (Name) expression;
+      if (name.name().equals("True")) {
+        return true;
+      } else {
+        Expression assignedValue = Expressions.singleAssignedValue(name);
+        return isArgumentTrue(assignedValue);
+      }
+    }
+    return false;
+  }
```

Comment:
```suggestion
  public static Predicate<Expression> isTrue() {
    return expression -> Optional.ofNullable(expression)
      .filter(Name.class::isInstance).map(Name.class::cast)
      .map(Name::name)
      .filter("True"::equals).isPresent();
  }
```
In the other CDK-related checks we used predicate to make the checks more readable. Maybe it's an option for you.-.